dvci/vaxx.link
# Custom Dataset for SHLink Creation

# Summary
Adds the ability for the user to create a SMART Health Link for a custom vaccine data set.

## New Behavior
The user can now create a new custom data set from all the available vaccines, and create a new sharing link for it. If the link is deleted, another one can be created since the data set will be saved to local storage.

## Code Changes
* New route “health-links/new-custom” for creating a custom dataset and then creating a SHLink for it
* Updates the reducer to add a new dataset 
* New helper function `filterForIds` to filter all the vaccines to those that should be included in a data set
* New component `SHCLinkCreateCustom` that displays all the vaccines, keeps track of check boxes for the vaccines, and allows for a new data set to be created along with a new SHLink for that dataset

# Testing Guidance
* Follow the [server setup instructions](https://github.com/dvci/vaxx.link/tree/main/server) from the vaxx.link/server directory. The server will run on port 8000.
* Run the UI with `npm start` from the vaxx.link/ui directory. The app will run on port 3000 and is accessible at http://localhost:3000/.
* Navigate to “Health Links” in the nav bar. There should be two default datasets already populated - All Vaccines and School Vaccines. There should be a third header that says “New Custom Data Set.” Click on the “Create new link” next to the “New Custom Data Set” header. 
* After clicking on the button, you should be navigated to a new page for selecting the vaccines to include in the new data set. Select some of the vaccines, then click “Activate New Sharing Link.”
* After clicking on the button, you will be brought back to the Health Links page. Check that a the custom data set has been created (its header defaults to Custom Data Set {data set id}.) A SHLink should be created for the data set. To check that the data set was appropriately filtered to just include the checked vaccines, try deactivating the SHLink and clicking on “Create new link” again. The options should be limited to those that were checked when creating the custom data set.
